File: flutter/packages/gifts/lib/src/domain/entities/gift.dart
import 'package:equatable/equatable.dart';

/// A gift in the catalog.
class Gift extends Equatable {
  final int id;
  final String name;
  final int type;
  final int? categoryId;
  final int price; // coins
  final String img;
  final String showImg;
  final String imageType;
  final int vipLevel;

  const Gift({
    required this.id,
    required this.name,
    required this.type,
    required this.categoryId,
    required this.price,
    required this.img,
    required this.showImg,
    required this.imageType,
    required this.vipLevel,
  });

  @override
  List<Object?> get props => [id];
}

File: backend/packages/gifts/src/Models/Gift.php
<?php

namespace Utd\Gifts\Models;

use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Support\Facades\Cache;

class Gift extends Model
{
    /** Gift type ids (mirror Eagle): 1 normal … 6 lucky. */
    public const TYPE_NORMAL = 1;
    public const TYPE_LUCKY = 6;

    /** Bust the cached image list whenever a gift changes. */
    protected static function booted(): void
    {
        $forget = fn () => Cache::forget('gifts:images');
        static::saved($forget);
        static::deleted($forget);
    }

    protected $fillable = [
        'name', 'e_name', 'type', 'gift_category_id', 'vip_level', 'price',
        'img', 'show_img', 'show_img2', 'image_type', 'music_gift', 'international_gift',
        'is_play', 'sort', 'enable', 'use_count',
    ];

    protected $casts = [
        'type'               => 'integer',
        'vip_level'          => 'integer',
        'price'              => 'integer',
        'music_gift'         => 'boolean',
        'international_gift'  => 'boolean',
        'is_play'            => 'boolean',
        'enable'             => 'boolean',
        'use_count'          => 'integer',
    ];

    public function category(): BelongsTo
    {
        return $this->belongsTo(GiftCategory::class, 'gift_category_id');
    }

    public function scopeEnabled(Builder $query): Builder
    {
        return $query->where('enable', true);
    }

    public function isLucky(): bool
    {
        return (int) $this->type === self::TYPE_LUCKY;
    }
}
